src/sentry/rules/base.py

```diff
@@ -37,14 +37,7 @@
 CallbackFuture = namedtuple("CallbackFuture", ["callback", "kwargs", "key"])
 
 
-class RuleDescriptor(type):
-    def __new__(cls, *args, **kwargs):
-        new_cls = super().__new__(cls, *args, **kwargs)
-        new_cls.id = f"{new_cls.__module__}.{new_cls.__name__}"
-        return new_cls
-
-
-class RuleBase(metaclass=RuleDescriptor):
+class RuleBase(abc.ABC):
```

Comment:
Should `id` be an required property here?

Reply:
Python 3.9 has a nice pattern for this:
```py
@classmethod
@property
def id(cls) -> str:
    raise NotImplemented
```
I'm not sure how to do this in 3.8 without messing with metaclasses.

Reply:
I think this works:
```
    @property
    @abstractmethod
    def my_abstract_property(self):
        ...
```

https://docs.python.org/3.8/library/abc.html#abc.abstractmethod

Reply:
I guess it's not a class level property but I think that's still fine